### DISCUSSION TITLE
How to open measurements file via python

### DISCUSSION THREAD

**Discussion by tinkeringtuck** [danielhrisca/asammdf#933] [Q&A]
in a terminal, I can open a measurements file via `asammdf --measurements <file_path>`. I want to do this programmatically via python. I see in the documentation you can plot individual signals, but I basically want to mimic the functionality of the above terminal command. I found the main() function from asammdf.gui.asammdfgui and tried the following:

```python
from asammdf.gui.asammdfgui import main
main('file_path')
```

A asammdf[gui] window opens but no channels/signals are shown like when running `asammdf --measurements <file_path>` from the terminal. Am I missing something obvious here or going about this the wrong way? 

Thanks.

**Comment by danielhrisca** [comment]
The argument for the `main` function has to be a list

```python
from asammdf.gui.asammdfgui import main
main(['file_path'])
```

**Reply by tinkeringtuck** [reply]
Thanks for the quick reply and correction. I wasn't sure if it was required or not, but I tried that and it yields the same result (see image below for window that opens. It opens a blank window, unlike the `asammdf --measurements <file_path>` command. Are you experiencing a different outcome?

![image](https://github.com/danielhrisca/asammdf/assets/64051003/365cdf79-59f3-4167-addf-89c28a42d63e)

**Reply by tinkeringtuck** [reply]
@danielhrisca do you have any additional insight on this?

**Reply by tinkeringtuck** [reply]
@danielhrisca sorry to pester but I wanted to update you that I did find the issue as well as a proposed solution, I will put together a PR shortly and include that below but I want to get up to speed on PR expectations and that may take me some time. Here is an example of trying to plot a measurements file via the command line and via main from asammdf.gui.asammdfgui, to illustrate the issue:

Command Line:
```
❯ asammdf --measurements 'test_2022-09-01-00_2022-09-02-00.mf4'
ldf is not supported
xls is not supported
xlsx is not supported
yaml is not supported
(Pdb) print(args.measurements)
['test_2022-09-01-00_2022-09-02-00.mf4']
(Pdb) print(measurements)
None
```

Python main module in asammdf.gui.asammdfgui
```
>>> from asammdf.gui.asammdfgui import main
ldf is not supported
xls is not supported 
xlsx is not supported
yaml is not supported
>>> main(['test_2022-09-01-00_2022-09-02-00.mf4'])
(Pdb) print(args.measurements)
None  
(Pdb) print(measurements)
['test_2022-09-01-00_2022-09-02-00.mf4']
```

I think there may be better ways to do it but my current proposed solution will be:
```python
if measurements:
    args.measurements = measurements
_main_window = MainWindow(args.measurements)
```

**Reply by danielhrisca** [reply]
@tinkeringtuck please try the latest development branch code

```python
from asammdf.gui.asammdfgui import main
main(['file_path'])
```

**Reply by tinkeringtuck** [reply]
reviewing your code, that implementation should work. I cloned the repo and was trying to run development branch but it I am running into other issues that I think are unrelated. All that to say, unless I can figure those out I don't plan to test it, but I think it should work as you implemented. Thanks. 

FWIW the issue I am running into is:

`ModuleNotFoundError: No module named 'asammdf.blocks.cutils'`

It looks like this is documented in other issues, which I reviewed, but wasn't totally clear what my solution should be. I am running Python 3.9.13 on Windows.

**Reply by danielhrisca** [reply]
You have to install not just overwrite

pip install -I --no-deps https://github.com/danielhrisca/asammdf/archive/development.zip

**Comment by tinkeringtuck** [comment]
Attempting that required the install of C++ build tools, but doing so allowed progress and I was able to successfully run

`pip install -I --no-deps https://github.com/danielhrisca/asammdf/archive/development.zip`

Unfortunately, now I am getting the following error:

```>>> from asammdf.gui.asammdfgui import main
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\jtuck\asammdf\.venv\lib\site-packages\asammdf\__init__.py", line 3, in <module>
    import logging
  File "C:\Program Files\Python39\lib\logging\__init__.py", line 26, in <module>
    import sys, os, time, io, re, traceback, warnings, weakref, collections.abc
  File "C:\Program Files\Python39\lib\re.py", line 124, in <module>
    import enum
  File "C:\Program Files\Python39\lib\enum.py", line 2, in <module>
    from types import MappingProxyType, DynamicClassAttribute
  File "C:\Users\jtuck\asammdf\src\asammdf\types.py", line 1, in <module>
    from bz2 import BZ2File
  File "C:\Program Files\Python39\lib\bz2.py", line 16, in <module>
    from threading import RLock
  File "C:\Program Files\Python39\lib\threading.py", line 6, in <module>
    import functools
  File "C:\Program Files\Python39\lib\functools.py", line 22, in <module>
    from types import GenericAlias
ImportError: cannot import name 'GenericAlias' from partially initialized module 'types' (most likely due to a circular import) (C:\Users\jtuck\asammdf\src\asammdf\types.py)```

**Reply by danielhrisca** [reply]
I haven't seen this so far
